### PR TITLE
ddl: allocate the element ID correctly for GC delete range

### LINKED ISSUE #67: Allocate the element ID correctly for GC delete range
After https://github.com/pingcap/tidb/pull/33726, the element ID is allocated locally for every job. 

For multi-schema change, it should be allocated locally at the job level, instead of the sub-job level.

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #67

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: parser/model/ddl.go [hunk 1]
```diff
@@ -265,7 +265,8 @@ type MultiSchemaInfo struct {
 	AddIndexes  []CIStr `json:"-"`
 	DropIndexes []CIStr `json:"-"`
 
-	RelativeColumns []CIStr `json:"-"`
+	RelativeColumns      []CIStr `json:"-"`
+	DeleteRangeElementID int     `json:"-"`
```
Comment (Defined2014) [inline parser/model/ddl.go]: Why add this param?

File: ddl/delete_range.go [hunk 2]
```diff
@@ -249,12 +267,11 @@ func (ea *elementIDAlloc) alloc() int64 {
 // insertJobIntoDeleteRangeTable parses the job into delete-range arguments,
 // and inserts a new record into gc_delete_range table. The primary key is
 // (job ID, element ID), so we ignore key conflict error.
-func insertJobIntoDeleteRangeTable(ctx context.Context, sctx sessionctx.Context, job *model.Job) error {
+func insertJobIntoDeleteRangeTable(ctx context.Context, sctx sessionctx.Context, job *model.Job, ea *elementIDAlloc) error {
```
Comment (Defined2014) [inline ddl/delete_range.go:261]: Why add `elementIDAlloc`?

Comment (tangenta) [reply ddl/delete_range.go:261]: Because the element ID should be unique in the parent job, instead of sub-job.

File: parser/model/ddl.go (hunk 1 above)
Comment (tangenta) [reply parser/model/ddl.go]: It was an accident :)

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**